pyanaconda/vnc.py

```diff
@@ -82,21 +82,22 @@ def __init__(self, root="/", ip=None, name=None,
 
     def setVNCPassword(self):
         """Set the vnc server password. Output to file. """
-
-        r, w = os.pipe()
         password_string = "%s\n" % self.password
-        os.write(w, password_string.encode("utf-8"))
 
-        with open(self.pw_file, "wb") as pw_file:
-            # the -f option makes sure vncpasswd does not ask for the password again
-            rc = util.execWithRedirect("vncpasswd", ["-f"],
-                                       stdin=r, stdout=pw_file,
-                                       binary_output=True, log_output=False)
+        # the -f option makes sure vncpasswd does not ask for the password again
+        proc = util.startProgram(
+            ["vncpasswd", "-f"],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE
+        )
 
-            os.close(r)
-            os.close(w)
+        out = proc.communicate(password_string.encode("utf-8"))[0]
 
-        return rc
+        if proc.returncode != 0:
+            raise OSError("Unable to set the VNC password.")
```

Comment:
You should also log stderr in this case please if possible.

Reply:
Fixed.